### PR TITLE
Feature/orthorex calculator

### PR DESCRIPTION
**Pull request by ebuka-nwabueze**

### CODE REVIEW

File: src/components/calculator/FinalAmount.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React, { FC } from 'react';
+import styled from 'styled-components';
+
+interface FinalAmountType {
+	data: {
+		productType: string;
```
Comment (Ozoemenamsolomon) [inline src/components/calculator/FinalAmount.tsx]: when you see yourself naming an attribute `somethingType` then you should consider using an enum, I think

File: src/components/styled/Forms.tsx [hunk 2]
```diff
@@ -41,3 +41,151 @@ export const StyledFormButtonControl = styled.div`
...
+		content: '';
+		width: 2em;
+		height: 2em;
+		border-radius: 50%;
+		transform: scale(1000);
+		transition: 120ms transform ease-in-out;
+		box-shadow: 0 0 0 1px var(--oex-orange);
+	}
+
+	&::checked::before {
```
Comment (Ozoemenamsolomon) [inline src/components/styled/Forms.tsx:66]: nice, just change one attribute on state change

File: src/pages/calculator.tsx [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+							<form action="">
+								<ParagraphText>
+									What is the shape of your project?
+								</ParagraphText>
+								<FormRadioWrapper>
+									<FormRadioGroup>
+										<FormRadioLabel>
+											<input
+												type="radio"
+												id="shape"
```
Comment (Ozoemenamsolomon) [inline src/pages/calculator.tsx]: your ids should be unique, compare them with ln100. That's the key for your `onInputChange` above

File: src/pages/calculator.tsx [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+import Link from 'next/link';
+import { useState } from 'react';
+import styled from 'styled-components';
+
+type OnChangeType =
+	| React.ChangeEvent<HTMLInputElement>
+	| React.ChangeEvent<HTMLSelectElement>;
+
+type FormDataType = {
+	shape: string;
```
Comment (Ozoemenamsolomon) [inline src/pages/calculator.tsx]: enum too, I guess

File: src/pages/calculator.tsx [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+	diameter: string;
```
Comment (Ozoemenamsolomon) [inline src/pages/calculator.tsx]: it should all be number, width, and length too

File: src/pages/calculator.tsx [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+}
+
+export default Calculator;
+
+const PageWrapper = styled.div`
+	background-color: var(--oex-lightest-grey);
+	margin-top: 5rem;
+	padding-top: 7rem;
+
+	@media (min-width: 768px) {
```
Comment (Ozoemenamsolomon) [inline src/pages/calculator.tsx]: start using the styled-component themed media query eg.  `@media ${({theme})=>theme.breakpoints.above.md}`

File: src/pages/composites/index.tsx [hunk 7]
```diff
@@ -72,7 +72,11 @@ const expectancies: ServiceCardType[] = [
 		description:
 			'Use our resin calculator to estimate the amount of epoxy resin you will need for your projects.',
 		image: calculator,
-		cta: <CTA white>Calculate your resin</CTA>,
+		cta: (
```
Comment (Ozoemenamsolomon) [inline src/pages/composites/index.tsx]: nice

File: src/styles/globals.css [hunk 8]
```diff
@@ -15,7 +15,6 @@ body {
 	--oex-lighter-grey: #e5e5e5;
 	--oex-lightest-grey: #f2f0f0;
 	--oex-orange: #fe7624;
-	--oex-light-orange: #fef9f5;
```
Comment (Ozoemenamsolomon) [inline src/styles/globals.css:18]: some of these variables look alike, they need to be reduced

**Review by Ozoemenamsolomon — COMMENTED** [review]
That's all for now

**Comment by Ozoemenamsolomon** [comment]
image
one of them should be selected by default, which in turn determines what's displayed

File: src/components/calculator/FinalAmount.tsx (hunk 1 above)
Comment (ebuka-nwabueze) [reply src/components/calculator/FinalAmount.tsx]: Ok. Used the enum type here.

File: src/pages/calculator.tsx (hunk 3 above)
Comment (ebuka-nwabueze) [reply src/pages/calculator.tsx]: What is In100? Should the id be something like shape100?

File: src/pages/calculator.tsx (hunk 4 above)
Comment (ebuka-nwabueze) [reply src/pages/calculator.tsx]: OK. Changed the type as well to enum.

File: src/pages/calculator.tsx (hunk 5 above)
Comment (ebuka-nwabueze) [reply src/pages/calculator.tsx]: Ok. Done

File: src/pages/calculator.tsx (hunk 3 above)
Comment (ebuka-nwabueze) [reply src/pages/calculator.tsx]: If the shape has a different id, i will use the name as the key, for radio buttons to work together, they must have the same name attribute

**Comment by ebuka-nwabueze** [comment]
> image one of them should be selected by default, which in turn determines what's displayed

If one is checked, the value does not change anymore.

File: src/utils/calculator.ts [hunk 9]
```diff
@@ -0,0 +1,99 @@
+import { UNITSTYPE } from 'pages/calculator';
+
+const centimetreToMetre = 0.01;
+const inchesToMetre = 0.0254;
+const feetToMetre = 0.3048;
+const densityOfResinInKg = 1110;
+
+const convertToMetre = (unit: string) => {
```
Comment (Ozoemenamsolomon) [inline src/utils/calculator.ts]: The parameter should be oof type `UnitType`

File: src/utils/calculator.ts [hunk 10]
```diff
@@ -0,0 +1,99 @@
+import { UNITSTYPE } from 'pages/calculator';
```
Comment (Ozoemenamsolomon) [inline src/utils/calculator.ts]: please only export the page component and the nextjs data fetrching funtions in the pages directory, every other export should be somewhere else eg. components, utils etc.

File: src/pages/calculator.tsx [hunk 11]
```diff
@@ -0,0 +1,386 @@
...
+} from '@components/styled/Forms';
+import Link from 'next/link';
+import { useState } from 'react';
+import styled from 'styled-components';
+import {
+	calculateCylinderResinInKg,
+	calculateRectangularResinInKg,
+} from 'utils/calculator';
+
+export enum PRODUCTTYPE {
```
Comment (Ozoemenamsolomon) [inline src/pages/calculator.tsx]: Please remove all these types export and take them somewhere else, maybe a separate `data` file or so

File: src/components/Header.tsx [hunk 12]
```diff
@@ -33,11 +33,14 @@ const Header: React.FC<HeaderProp> = ({ pathname }) => {
 	const [light, setLight] = useState(false);
 
 	useEffect(() => {
+<<<<<<< HEAD
```
Comment (Ozoemenamsolomon) [inline src/components/Header.tsx]: please look into this theonly change should be the addition to calculator route

**Review by Ozoemenamsolomon — CHANGES_REQUESTED** [review]
👍🏽

File: src/pages/calculator.tsx (hunk 11 above)
Comment (Ozoemenamsolomon) [reply src/pages/calculator.tsx]: `FinalAmount.tsx` would ba a better option

File: src/pages/calculator.tsx [hunk 13]
```diff
@@ -0,0 +1,386 @@
...
+	OEX5302 = 'OEX5302',
+	OEX5311 = 'OEX5311',
+}
+export enum SHAPETYPE {
```
Comment (Ozoemenamsolomon) [inline src/pages/calculator.tsx:23]: You exported thisb but it's not used anywhere else

Comment (Ozoemenamsolomon) [inline src/pages/calculator.tsx]: You exported this but it's not used anywhere else

File: src/utils/calculator.ts (hunk 9 above)
Comment (ebuka-nwabueze) [reply src/utils/calculator.ts]: Done

File: src/utils/calculator.ts (hunk 10 above)
Comment (ebuka-nwabueze) [reply src/utils/calculator.ts]: Ok, I have transfered the defintion to the utils

File: src/components/styled/Forms.tsx (hunk 2 above)
Comment (ebuka-nwabueze) [reply src/components/styled/Forms.tsx:66]: Could you point out here what you meant?

File: src/pages/calculator.tsx (hunk 6 above)
Comment (ebuka-nwabueze) [reply src/pages/calculator.tsx]: Ok. Done

File: src/styles/globals.css (hunk 8 above)
Comment (ebuka-nwabueze) [reply src/styles/globals.css:18]: I won't add anymore colors, i will use only these ones. I'm trying to get the design

File: src/pages/calculator.tsx (hunk 11 above)
Comment (ebuka-nwabueze) [reply src/pages/calculator.tsx]: Sure done

File: src/components/Header.tsx (hunk 12 above)
Comment (ebuka-nwabueze) [reply src/components/Header.tsx]: Yeah, i fixed it in the file.

File: src/pages/calculator.tsx (hunk 13 above)
Comment (ebuka-nwabueze) [reply src/pages/calculator.tsx]: moved it to the utils file

File: src/components/styled/Forms.tsx (hunk 2 above)
Comment (Ozoemenamsolomon) [reply src/components/styled/Forms.tsx:66]: No attention is needed, just commending

File: src/styles/globals.css (hunk 8 above)
Comment (Ozoemenamsolomon) [reply src/styles/globals.css:18]: Alright, consider adding them only if there's a very obvious visual difference from an already existing hex

File: src/pages/calculator.tsx [hunk 14]
```diff
@@ -0,0 +1,455 @@
...
+const ResinProductTitle = styled.h4`
+	font-size: 1.2rem;
+	margin: 0;
+
+	@media (${({ theme }) => theme.breakpoints.above.md}) {
+		font-size: 2rem;
+	}
+`;
+
+export const ProductCards = styled.div`
```
Comment (Ozoemenamsolomon) [inline src/pages/calculator.tsx]: move this to a styled folder, only export the page or the data fetching functions from a page directory.

File: src/pages/calculator.tsx (hunk 3 above)
Comment (Ozoemenamsolomon) [reply src/pages/calculator.tsx]: line 100

Comment (Ozoemenamsolomon) [reply src/pages/calculator.tsx]: consider using the name in your onChange then, instead of ID, so that ID would be unique in a page

**Comment by Ozoemenamsolomon** [comment]
remember to set the PR as ready, when you think it's ready

File: src/pages/calculator.tsx (hunk 14 above)
Comment (ebuka-nwabueze) [reply src/pages/calculator.tsx]: Ok. I won't need the export again. So i removed the export

**Review by Ozoemenamsolomon — APPROVED** [review]
LGTM but the components in `src/components/styled/Forms.tsx` need to be looked into

**Pull request merged**